nexus/src/app/background/blueprint_load.rs

```diff
@@ -0,0 +1,159 @@
+use futures::future::BoxFuture;
+use futures::FutureExt;
+use nexus_db_queries::context::OpContext;
+use nexus_db_queries::db::DataStore;
+use nexus_types::deployment::Blueprint;
+use serde_json::json;
+use std::sync::Arc;
+use tokio::sync::watch;
+
+pub struct TargetBlueprintLoader {
+    datastore: Arc<DataStore>,
+    last: Option<Arc<Blueprint>>,
+    tx: watch::Sender<Option<Arc<Blueprint>>>,
+    rx: watch::Receiver<Option<Arc<Blueprint>>>,
+}
+
+impl TargetBlueprintLoader {
+    pub fn new(datastore: Arc<DataStore>) -> TargetBlueprintLoader {
+        let (tx, rx) = watch::channel(None);
+        TargetBlueprintLoader { datastore, last: None, tx, rx }
+    }
+
+    /// Expose the target blueprint
+    pub fn watcher(&self) -> watch::Receiver<Option<Arc<Blueprint>>> {
+        self.rx.clone()
+    }
+}
+
+impl BackgroundTask for TargetBlueprintLoader {
+    fn activate<'a>(
+        &'a mut self,
+        opctx: &'a OpContext,
+    ) -> BoxFuture<'a, serde_json::Value> {
+        async {
+            // Set up a logger for this activation that includes metadata about
+            // the current target.
+            let log = match &self.last {
+                None => opctx.log.clone(),
+                Some(old) => opctx.log.new(o!(
+                    "current_target_id" => old.id.to_string(),
+                    "current_time_created" => old.time_created.to_string(),
+                )),
+            };
+
+            // Retrieve the latest target blueprint
+            let result =
+                self.datastore.blueprint_target_get_current_full(opctx).await;
+
+            // Decide what to do with the result
+            match (&self.last, result) {
+                (_, Err(error)) => {
+                    // We failed to read the blueprint. There's nothing to do
+                    // but log an error. We'll retry when we're activated again.
+                    let message = format!("{:#}", error);
+                    warn!(
+                        &log,
+                        "failed to read target blueprint";
+                        "error" => &message
+                    );
+                    json!({
+                        "error": 
+                        format!("failed to read target blueprint: {message}")})
+                }
+                (None, Ok(None)) => {
+                    // We haven't found a blueprint yet. Do nothing.
+                    json!({})
+                }
+                (Some(old), Ok(None)) => {
+                    // We have transitioned from having a blueprint to not
+                    // having one. This should not happen.
+                    let message = format!(
+                        "target blueprint with id {} was removed. There is no \
+                        longer any target blueprint",
+                        old.id
+                    );
+                    error!(&log, "{}", message);
+                    json!({"error": message})
```

Comment:
Agreed this shouldn't happen.  If it did, I wonder if we should set the current one to `None` anyway?  I'm not really sure to be honest but if Nexus restarted after this point, that's what would happen anyway.

If we're going to keep the existing one, I feel like we should include that information in the JSON object too.

Reply:
Good idea. Done in [596c9be](https://github.com/oxidecomputer/omicron/pull/4891/commits/596c9be466c0ba54c41201edf0c18d9b8c6c6280)

Reply:
Whoops, meant to add a `status` field here as done elsewhere. Will do in next commit.

Reply:
Added more details in `[5d72471](https://github.com/oxidecomputer/omicron/pull/4891/commits/5d72471ad30c517f19c94d1eab9274fde74bf61d)